compiler/wasm/src/compile.rs

```diff
@@ -128,16 +128,21 @@ impl JsCompileContractResult {
 #[derive(Deserialize, Default)]
 pub(crate) struct DependencyGraph {
     pub(crate) root_dependencies: Vec<CrateName>,
-    pub(crate) library_dependencies: HashMap<CrateName, Vec<CrateName>>,
+    pub(crate) library_dependencies: BTreeMap<CrateName, Vec<CrateName>>,
 }
 #[wasm_bindgen]
-// This is a map containing the paths of all of the files in the entry-point crate and
-// the transitive dependencies of the entry-point crate.
-//
-// This is for all intents and purposes the file system that the compiler will use to resolve/compile
-// files in the crate being compiled and its dependencies.
+/// This is a map contains the paths of all of the files in the entry-point crate and
+/// the transitive dependencies of the entry-point crate.
+///
+/// This is for all intents and purposes the file system that the compiler will use to resolve/compile
+/// files in the crate being compiled and its dependencies.
+///
+/// Using a `BTreeMap` to add files to the [FileManager] in a deterministic order,
+/// which affects the `FileId` in the `Location`s in the AST on which the `hash` is based.
+/// Note that we cannot expect to match the IDs assigned by the `FileManager` used by `nargo`,
+/// because there the order is determined by the dependency graph as well as the file name.
 #[derive(Deserialize, Default)]
-pub struct PathToFileSourceMap(pub(crate) HashMap<std::path::PathBuf, String>);
+pub struct PathToFileSourceMap(pub(crate) BTreeMap<std::path::PathBuf, String>);
```

Comment:
Can you move the `#[wasm_bindgen]` down below the doc comment to avoid it getting lost?